modules/ding_reservation/ding_reservation.module

```diff
@@ -185,7 +185,16 @@ function ding_reservation_reserve_ajax($entity, $reservable = NULL) {
       // the options to select branch and period.
       $defaults = ding_provider_invoke('reservation', 'default_options', $user);
       $matches = preg_grep("/preferred_branch$/", array_keys($defaults));
-      if (empty($defaults[array_shift($matches)])) {
+      $profile = ding_user_provider_profile($user);
+      $user_quick_reservation = $profile->field_quick_reservation[LANGUAGE_NONE][0]['value'];
+      $admin_quick_reservation = variable_get('ding_quick_reservation');
+      if ($admin_quick_reservation == "0") {
+        $quick_reservation = ($user_quick_reservation == "0") ? TRUE : FALSE;
+      }
+      else {
+        $quick_reservation = ($user_quick_reservation == "0") ? FALSE : TRUE;
+      }
+      if (empty($defaults[array_shift($matches)]) || $quick_reservation) {
```

Comment:
Better avoid complex nested calls inside conditionals.
If this is split to variables it's more easier to understand and debug.